src/modules/flow/iio/nodes.c

```diff
@@ -153,27 +160,30 @@ gyroscope_open(struct sol_flow_node *node, void *data, const struct sol_flow_nod
     return -EINVAL;
 }
 
+
 static void
-gyroscope_close(struct sol_flow_node *node, void *data)
+iiodevice_close(struct sol_flow_node *node, void *data)
 {
-    struct gyroscope_data *mdata = data;
+    struct iiodevice_data *mdata = data;
 
     free((char *)mdata->config.trigger_name);
     if (mdata->device)
         sol_iio_close(mdata->device);
 }
 
 static int
-gyroscope_tick(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+iiodevice_tick(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
 {
     static const char *errmsg = "Could not read channel values";
-    struct gyroscope_data *mdata = data;
+    struct iiodevice_data *mdata = data;
+
+    SOL_WRN("%s mdata=%p \n", __func__, mdata);
 
     if (mdata->buffer_enabled) {
         if (!sol_iio_device_trigger_now(mdata->device))
             goto error;
     } else {
-        reader_cb(node, mdata->device);
+        mdata->config.sol_iio_reader_cb(node, mdata->device);
```

Comment:
```c
const struct iio_node_type *type = (const struct iio_node_type *)sol_flow_node_get_type(node);
type->read_cb(node, mdata->device);
```

This will work once you define your `iio_node_type` extending `sol_flow_node_type`. See how it's done in http-server:
 - https://github.com/solettaproject/soletta/blob/master/src/modules/flow/http-server/http-server.c#L74 (struct declaration)
 - https://github.com/solettaproject/soletta/blob/master/src/modules/flow/http-server/http-server.json#L22 (json declaration)
 - https://github.com/solettaproject/soletta/blob/master/src/modules/flow/http-server/http-server.c#L234 (fetch and cast)